compiler-plugin/src/main/kotlin/dev/nhachicha/AccessorModifierPlugin.kt

```diff
@@ -54,55 +63,93 @@ class AccessorModifierComponentRegistrar : ComponentRegistrar {
-                        return IrBlockBuilder(context, currentScope?.scope!!, expression.startOffset, expression.endOffset).irBlock {
-                            val irConcat = irConcat()
-                            irConcat.addArgument(irString("Hello "))
-                            irConcat.addArgument(expression.value)
-                            +irReturn(irConcat)
+    companion object {
+        fun generate(
+                irClass: IrClass,
+                context: IrPluginContext
+        ) {
+            if (irClass.isRealmObject) {
+                irClass.transformChildrenVoid(object : IrElementTransformerVoidWithContext() {
+                    override fun visitFunctionNew(declaration: IrFunction): IrStatement {
+                        return if (declaration.isPropertyAccessor
+                                && declaration.isGetter
+                                && (declaration.returnType.isString() || declaration.returnType.isNullableString())
+                        ) {
+                            declaration.body?.transformChildrenVoid(object : IrElementTransformerVoid() {
+                                override fun visitReturn(expression: IrReturn): IrExpression {
+                                    logger("modifying expression: ${expression.dump()}")
+                                    return IrBlockBuilder(context, currentScope?.scope!!, expression.startOffset, expression.endOffset).irBlock {
+                                        val irConcat = irConcat()
+                                        irConcat.addArgument(irString("Hello "))
+                                        irConcat.addArgument(expression.value)
+                                        +irReturn(irConcat)
+                                    }
+                                }
+                            })
+                            super.visitFunctionNew(declaration)
+                        } else {
+                            super.visitFunctionNew(declaration)
                         }
                     }
                 })
-                super.visitFunctionNew(declaration)
-            } else {
-                super.visitFunctionNew(declaration)
             }
         }
-
     }
 }
 
-fun FileLoweringPass.runOnFileInOrder(irFile: IrFile) {
+fun ClassLoweringPass.runOnFileInOrder(irFile: IrFile) {
     irFile.acceptVoid(object : IrElementVisitorVoid {
         override fun visitElement(element: IrElement) {
             element.acceptChildrenVoid(this)
         }
 
-        override fun visitFile(declaration: IrFile) {
+        override fun visitClass(declaration: IrClass) {
             lower(declaration)
             declaration.acceptChildrenVoid(this)
         }
     })
 }
+
+// Annotation
+private val IrClass.isRealmObject: Boolean get() = kind == ClassKind.CLASS && hasRealmObjectAnnotationWithoutArgs()
+
+private val realmObjectAnnotationFqName = FqName("io.realm.kmmapplication.shared.RealmObject")
```

Comment:
I guess we should expose the annotation as part of the plugin.

Reply:
I think you would inject these kind of dependencies by adding them to the project by overriding the `apply(target: Project)` in the gradle plugins implementation of `KotlinCompilerPluginSupportPlugin`.